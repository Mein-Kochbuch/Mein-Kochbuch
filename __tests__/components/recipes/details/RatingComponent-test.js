import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import RatingComponent from '../../../../src/components/recipes/details/RatingComponent';
import {Rating} from 'react-native-ratings';
import Icon from 'react-native-vector-icons/Feather';

import Colors from 'react-native/Libraries/NewAppScreen/components/Colors';

describe('RatingComponent Test', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Test light theme', () => {
    const averageRating = 4;
    const ratingCount = 12;
    const ownRating = 2.5;
    const handleRating = () => {};

    const component = renderer.create(
      <RatingComponent
        averageRating={averageRating}
        ratingCount={ratingCount}
        ownRating={ownRating}
        handleRating={handleRating}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.toJSON().children.length).toBe(4);
    expect(component.toJSON().children[1].children[0]).toBe(
      averageRating.toString(),
    );
    expect(component.toJSON().children[3].children[0]).toBe(`(${ratingCount})`);

    const testInstance = component.root;

    expect(testInstance.findAllByType(Rating)[0].props.type).toBe('custom');
    expect(testInstance.findAllByType(Rating)[0].props.fractions).toBe(1);
    expect(testInstance.findAllByType(Rating)[0].props.imageSize).toBe(30);
    expect(testInstance.findAllByType(Rating)[0].props.startingValue).toBe(
      ownRating,
    );
    expect(testInstance.findAllByType(Rating)[0].props.tintColor).toBe(
      Colors.lighter,
    );
    expect(testInstance.findAllByType(Rating)[0].props.onFinishRating).toBe(
      handleRating,
    );

    expect(testInstance.findAllByType(Icon)[0].props.name).toBe('star');
    expect(testInstance.findAllByType(Icon)[0].props.size).toBe(20);
    expect(testInstance.findAllByType(Icon)[0].props.color).toBe('#F1C40F');
  });
});
