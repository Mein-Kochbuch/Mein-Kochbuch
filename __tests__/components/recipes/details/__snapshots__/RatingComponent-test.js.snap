// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiListItem Test RatingComponent Test dark 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    pointerEvents="auto"
    style={
      Array [
        Object {
          "paddingTop": 12,
          "width": 200,
        },
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#f1c40f",
              "height": 30,
              "width": 75,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "white",
              "height": 30,
              "width": 75,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#222",
              "width": 30,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#222",
              "width": 30,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#222",
              "width": 30,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#222",
              "width": 30,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#222",
              "width": 30,
            }
          }
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "paddingTop": 12,
        },
      ]
    }
  >
    4
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#F1C40F",
          "fontSize": 20,
        },
        Array [
          Object {
            "paddingTop": 12,
          },
        ],
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 10,
          "paddingTop": 12,
        },
      ]
    }
  >
    (12)
  </Text>
</View>
`;

exports[`MultiListItem Test RatingComponent Test light 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    pointerEvents="auto"
    style={
      Array [
        Object {
          "paddingTop": 12,
          "width": 200,
        },
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#f1c40f",
              "height": 30,
              "width": 75,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "white",
              "height": 30,
              "width": 75,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#F3F3F3",
              "width": 30,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#F3F3F3",
              "width": 30,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#F3F3F3",
              "width": 30,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#F3F3F3",
              "width": 30,
            }
          }
        />
      </View>
      <View>
        <Image
          source={
            Object {
              "testUri": "../../react-native-ratings/dist/images/star.png",
            }
          }
          style={
            Object {
              "height": 30,
              "tintColor": "#F3F3F3",
              "width": 30,
            }
          }
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "paddingTop": 12,
        },
      ]
    }
  >
    4
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#F1C40F",
          "fontSize": 20,
        },
        Array [
          Object {
            "paddingTop": 12,
          },
        ],
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 10,
          "paddingTop": 12,
        },
      ]
    }
  >
    (
    12
    )
  </Text>
</View>
`;
